### PR TITLE
Add clarification comment to proxy.c::on_body

### PR DESCRIPTION
**Pull request by hfujita**

### CODE REVIEW

**Review by deweerdt — APPROVED** [review]
thank you!

**Comment by kazuho** [comment]
Thank you for the pull request.

While the explanation is technically correct, I wonder if it is verbose. Hence the question: why do you think that we need an explanation here?

Is is the trouble of having the understanding that `generator_disposed` meaning that `self` has been disposed of? FWIW, `self` (of type `rp_generator_t`) is a subtype of `h2o_generator_t`, therefore "self" and "generator" are synonyms here.

If so, does renaming `generator_disposed` to something like `self_disposed` resolve your concern?

**Comment by hfujita** [comment]
I think my biggest struggle at the initial glance was that it was not clear to me that `on_body_on_close` could free `self`.  What if I simplify the comment to a one-liner like `/* on_body_on_close could free self */`?

File: lib/core/proxy.c [hunk 1]
```diff
@@ -491,6 +491,9 @@ static int on_body(h2o_httpclient_t *client, const char *errstr)
     if (errstr != NULL) {
         self->generator_disposed = &generator_disposed;
         on_body_on_close(self, errstr);
+        /* At this point, the generator (`self`) may have been disposed and therefore untouchable.
+         * If that is the case, `generator_disposed` becomes true, and since `self` no longer exists,
+         * there is no risk of `&generator_disposed` leaking outside of this function. */
```
Comment (kazuho) [inline lib/core/proxy.c:496]:
```suggestion
        /* Call `on_body_on_close`. This function might dispose `self`, in which case `generator_disposed` would be set to true. */
        self->generator_disposed = &generator_disposed;
        on_body_on_close(self, errstr);
```

**Review by kazuho — COMMENTED** [review]
@hfujita I see. Maybe something like:

**Comment by hfujita** [comment]
@kazuho Looks good, thank you for the suggestion!

**Pull request merged**